providers/azure.go

```diff
@@ -88,9 +81,21 @@ func NewAzureProvider(p *ProviderData, opts options.AzureOptions) *AzureProvider
 		overrideTenantURL(p.RedeemURL, azureDefaultRedeemURL, tenant, "token")
 	}
 
+	groupField := azureDefaultGroupField
+	if opts.GroupField != "" {
+		groupField = opts.GroupField
+	}
+
+	isV2Endpoint := false
+	if strings.Contains(p.LoginURL.String(), "v2.0") {
```

Comment:
instead of having a check like this, does it make sense to just have an option in the azure provider config to specify v1 or v2, and then have default v1 and v2 URL's? I think that would simplify the setup for new users who would like to use v2 endpoints.

Reply:
Hmm, not sure how we can correlate the version option with an explicit `--oidc-issuer-url` option a user can set. What happens in a scenario where a user set the version option to V2, but passes the oidc url associated with V1?

Reply:
I'm just thinking in terms making the Azure provider entry level super simple - that example you provided is advanced and I would expect the user to know the difference between v1 and v2 azure issuers in that case.

Reply:
If the current approach become a pain point, we can consider adjusting it a followup PR.